pkg/distributor/distributor.go

```diff
@@ -1505,6 +1507,29 @@ func forReplicationSet[T any](ctx context.Context, d *Distributor, replicationSe
 	return ring.DoUntilQuorum(ctx, replicationSet, d.queryQuorumConfig(ctx, replicationSet), wrappedF, cleanup)
 }
 
+// forReplicationSets runs f, in parallel, for all ingesters in the input replication sets.
+func forReplicationSets[R any](ctx context.Context, d *Distributor, replicationSets []ring.ReplicationSet, f func(context.Context, ingester_client.IngesterClient) (R, error)) ([]R, error) {
```

Comment:
Note to reviewers: this function contains some duplicate code from `forReplicationSet()`, but `forReplicationSet()` will disappear in few PRs.